### PR TITLE
[Transaction] Optimize transaction timeout

### PR DESCRIPTION
**Pull request by liangyepianzhou**
link https://github.com/apache/pulsar/issues/14166
### Motivation
A timeout is create every time a transaction is opened.
When there are a large number of tasks with long timeout and there is no cancel mechanism, there is a risk of OOM.
### Modification
Cancel the timeout of the completed transaction.

### Verifying this change

- [ ] Make sure that the change passes the CI checks.

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Does this pull request potentially affect one of the following parts:

*If `yes` was chosen, please highlight the changes*

  - Dependencies (does it add or upgrade a dependency): (yes / no)
  - The public API: (yes / no)
  - The schema: (yes / no / don't know)
  - The default values of configurations: (yes / no)
  - The wire protocol: (yes / no)
  - The rest endpoints: (yes / no)
  - The admin cli options: (yes / no)
  - Anything that affects deployment: (yes / no / don't know)

### Documentation

Check the box below or label this PR directly (if you have committer privilege).

Need to update docs? 

- [ ] `doc-required` 
  
  (If you need help on updating docs, create a doc issue)
  
- [x] `no-need-doc` 
  
  (Please explain why)
  
- [ ] `doc` 
  
  (If this PR contains doc changes)

### CODE REVIEW

**Review by codelipenghui — COMMENTED** [review]
LGTM, Could you please add a test for the changes?

File: pulsar-client/src/main/java/org/apache/pulsar/client/impl/transaction/TransactionImpl.java [hunk 1]
```diff
@@ -164,6 +168,7 @@ public synchronized void registerCumulativeAckConsumer(ConsumerImpl<?> consumer)
         return checkIfOpenOrCommitting().thenCompose((value) -> {
             CompletableFuture<Void> commitFuture = new CompletableFuture<>();
             this.state = State.COMMITTING;
+            timeout.cancel();
```
Comment (Jason918) [inline pulsar-client/src/main/java/org/apache/pulsar/client/impl/transaction/TransactionImpl.java:171]: Why cancel timeout task here, not after state is in `COMMITTED, ABORTED, ERROR` which this transaction is actually completed.

Comment (liangyepianzhou) [reply pulsar-client/src/main/java/org/apache/pulsar/client/impl/transaction/TransactionImpl.java:171]: The function of timeout is to set the status of the transaction to timeout when the transaction times out.
If the state of the transaction has become committing or aborting, the timeout will have no effect. At this point, the resources of this timetask can be released

**Review by Jason918 — APPROVED** [review]

**Review by codelipenghui — APPROVED** [review]

**Pull request merged**